### ISSUE TITLE
Memory leak observed with  Azure Cosmos DB Java SDK v4 for Core (SQL)

### ISSUE THREAD

**Issue by prasgarg** [Azure/azure-sdk-for-java#27765]
We are seeing from heap dumps multiple channelpools being created for a given fqdn when we are doing storedproc exec at 50rps using cosmos Sync Client and gateway mode. To me it seems like they should be re-used and there should be just 1 channel pool per fqdn.  When our container died from memory exhaustion it had 180k such channel pools. 
There seems to be an existing issue similar to this https://github.com/reactor/reactor-netty/issues/1472 which is marked closed can you please investigate.

**Comment by prasgarg** [comment]
We are already using https://mvnrepository.com/artifact/com.azure/azure-cosmos/4.25.0 but we have tried it on 4.27.0 as well and the issue exists on latest version as well. This is reproducible every time i run a load for around an hour. 

For the repro I can give a heap dump if that helps. The heap dump shows 2gb of memory all consumed by channelpools.

**Comment by anuchandy** [comment]
1. The cosmos-SDK component uses a _specific_ version of the reactor, reactor-netty, and netty libraries. Do you have any other application component that brings in one of these libraries but a different version from what Cosmos-SDK defines? If yes, can you check the same memory pattern exists if we align the versions? i.e., bumping up the other components-version such that they use the same version of these libraries as cosmos-SDK. 
2. Like Kushagra mentioned, If you can come up with a minimal java-project that reproduces the case, that will speed up the investigation.

**Comment by prasgarg** [comment]
We don't have any other application component bringing in these libraries so they are coming from cosmos sdk only. 
For us we were only observing this on azul jvm (we observed for java11 but didn't try other java versions)
After moving to oracle jvm we didn't observe this issue anymore even for continuous runs for 3+ days, it is still reproducible on azul jvm though within couple of hours. 
I will try to come up with the sample application next week but since we are already unblocked by jvm change we have not prioritized this

**Comment by allenliang2022** [comment]
Any update please on this memory issue please? Our team also hit the similar memory leak issue when we switch from gateway mode to direct connect mode. Please update the status of this issue, thanks.

**Comment by kushagraThapar** [comment]
@allenliang2022 / @prasgarg  - as mentioned earlier, this has already been fixed long time back in reactor-netty library and on azure-cosmos java SDK. 
If you are still experiencing issues, please move to the latest version of the azure-cosmos Java SDK - 4.29.1 and open a new ticket. 
@prasgarg  - I believe the issue that you are seeing is because of a proxy handler between your application and cosmos db endpoints.
